### PR TITLE
contributing: Is your PR easy to say "yes" to?

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

There are a number of common things I see in PRs that make them harder to review. I end up having to repeat myself when they happen. It would be convenient to have those things enumerated somewhere that I can point to.

#### Summary of Changes

Add a section to `CONTRIBUTING.md` that describes things that make PRs *harder* (and *easier*) to review.

[Rendered view](https://github.com/solana-labs/solana/blob/ae42646828e8f745e9d545c25216d23651f25199/CONTRIBUTING.md#is-your-pr-easy-to-say-yes-to)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -194,6 +195,35 @@ a commit.  After a reviewer adds feedback, they won't be checking on the status
...
+* Renaming variables/functions/types unnecessarily and/or without explanation.
+* Not following established conventions in the function/module/crate/repo.
+* Changing whitespace: moving code and/or reformatting code.
+* Force-pushing the branch unnecessarily; this makes it harder to track any
+  previous comments on specific lines of code, and also harder to track changes
+  already reviewed from previous commits.
+  * When force-pushing is required—for example to handle a merge conflict—and
+    no new changes have been made since the previous review, indicating as such
+    is beneficial.
+
```
Comment (mvines) [inline CONTRIBUTING.md:216]: * Not responding to comments from previous rounds of review. If feedback has been addressed, say so and resolve the comment.
* Requesting review too early. Is CI green? If the computer doesn't like your PR yet why would a human.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -194,6 +195,35 @@ a commit.  After a reviewer adds feedback, they won't be checking on the status
...
 
+### Is your PR easy to say "yes" to?
+
+PRs that are easier to review are more like to be reviewed. Strive to make your
+PR easy to say "yes" to.
+
+Non-exhaustive list of things that make it *harder* to review:
+
+* Additional changes that are orthogonal to the problem statement and proposed
+  changes.
```
Comment (mvines) [inline CONTRIBUTING.md:206]:
```suggestion
* Additional changes that are orthogonal to the problem statement and proposed
  changes. Instead move those changes to a different PR
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -194,6 +195,35 @@ a commit.  After a reviewer adds feedback, they won't be checking on the status
...
+* Renaming variables/functions/types unnecessarily and/or without explanation.
+* Not following established conventions in the function/module/crate/repo.
+* Changing whitespace: moving code and/or reformatting code.
```
Comment (mvines) [inline CONTRIBUTING.md:209]:
```suggestion
* Changing whitespace: moving code and/or reformatting code. Make such changes in a separate PR
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -194,6 +195,38 @@ a commit.  After a reviewer adds feedback, they won't be checking on the status
 of that feedback after every new commit. Instead, directly mention the reviewer
 when you feel your PR is ready for another pass.
 
+### Is your PR easy to say "yes" to?
+
+PRs that are easier to review are more like to be reviewed. Strive to make your
```
Comment (apfitzge) [inline CONTRIBUTING.md:200]: typo?
```suggestion
PRs that are easier to review are more likely to be reviewed. Strive to make your
```

**Review by apfitzge — COMMENTED** [review]
Not sure if this is a typo or my grammar is bad. Otherwise like a good list of suggestions to me

**Review by apfitzge — APPROVED** [review]
I like it!

**Pull request merged**